lib/main.js

```diff
@@ -245,35 +249,61 @@ var reporterPanel = panel.Panel({
     }
 });
 
+function retrieveAddonIcon(addon, callback) {
+    if (!addon.iconURL) {
+        addon.icon = genericAddonIconURL;
+        callback(addon);
+    } else {
+        var xhr = new XMLHttpRequest();
+        xhr.onreadystatechange = function () {
+            if (this.readyState === 4) {
+            addon.icon = 'data:image/png;base64,' + base64.encode(String.fromCharCode.apply(null, new Uint8Array(this.response)));
+            callback(addon);
+            }
+        }
+        xhr.open('GET', addon.iconURL, true);
+        xhr.responseType = 'arraybuffer';
+        xhr.send(null);
+    }
+}
+
 reporterPanel.sendAddonReports = function() {
     AddonManager.getAllAddons(function(addons) {
         var addonReports = [];
+        var reportsProcessed = 0;
+        var addonsCount = 0;
         for (var i=0; i<addons.length; i++) {
             if (addons[i].type === "extension" && !addons[i].isSystem) {
-            var addonReport = ACR.AddonReportStorage.getAddonReportByAddon(addons[i]);
-            addonReport.iconURL = (addons[i].iconURL?addons[i].iconURL:genericAddonIconURL);
-            if (!addons[i].isActive && addonReport.state == 0)
-                addonReport.state = 3;
-            if (addonReport.state == 0)
-                addonReport.state = 1;
-            addonReport.isDisabled = !addons[i].isActive;
-            addonReport.multiprocessCompatible = addons[i].multiprocessCompatible;
-            addonReports.push(addonReport);
+                addonsCount++;
+                retrieveAddonIcon(addons[i], function callback(addon) {
+                    var addonReport = ACR.AddonReportStorage.getAddonReportByAddon(addon);
+                    addonReport.icon = addon.icon;
+                    if (!addon.isActive && addonReport.state == 0)
+                        addonReport.state = 3;
+                    if (addonReport.state == 0)
+                        addonReport.state = 1;
+                    addonReport.isDisabled = !addon.isActive;
+                    addonReport.multiprocessCompatible = addon.multiprocessCompatible;
+                    addonReports.push(addonReport);
+                    reportsProcessed++;
+
+                    if (reportsProcessed === addonsCount) {
+                        var by = addonReports.length*55;
```

Comment:
Oh, I just moved that whole pile of code. I am not touching that since the whole panel sizing is not trivial.